### PR TITLE
MMU maintenance: Sync with 32-bit FW a little bit

### PR DESCRIPTION
**Pull request by gudnimg**
The changes in this commit are mainly formatting, but there are some changes like adding a new function `planner_draining()`.

There are more differences between 8-bit and 32-bit, but it's nice to get most of the formatting out of the way to make future PRs easier to review.

### CODE REVIEW

**Review by 3d-gussner — APPROVED** [review]
Looks good to me

File: Firmware/mmu2.cpp [hunk 1]
```diff
@@ -860,7 +885,7 @@ StepStatus MMU2::LogicStep(bool reportErrors) {
     case Finished:
         // At this point it is safe to trigger a runout and not interrupt the MMU protocol
         CheckFINDARunout();
-        break;
+        [[fallthrough]]; // let Finished be reported the same way like Processing
```
Comment (DRracer) [inline Firmware/mmu2.cpp]: Please verify this change on MK3S. It may break current implementation on this platform. MK4 may have a different workflow.

**Review by DRracer — APPROVED** [review]
Conditionally approved, just check the one functional change marked in my review. Otherwise ok

File: Firmware/mmu2.cpp (hunk 1 above)
Comment (3d-gussner) [reply Firmware/mmu2.cpp]: @gudnimg Did you test this?

Comment (gudnimg) [reply Firmware/mmu2.cpp]: No I don’t think I tested it specifically. We can drop this change if you think its risky.

**Pull request merged**